packages/core/src/auth/sso/ssoAccessTokenProvider.ts

```diff
@@ -67,16 +68,18 @@ export abstract class SsoAccessTokenProvider {
     public constructor(
         protected readonly profile: Pick<SsoProfile, 'startUrl' | 'region' | 'scopes' | 'identifier'>,
         protected readonly cache = getCache(),
-        protected readonly oidc: OidcClient = OidcClient.create(profile.region)
+        protected readonly oidc: OidcClient = OidcClient.create(profile.region),
+        protected readonly reAuthState: ReAuthState = ReAuthState.instance
     ) {}
 
-    public async invalidate(): Promise<void> {
+    public async invalidate(reason: string): Promise<void> {
```

Comment:
would it be possible to pass in a type that contains all the possible options? I think it might help to see all the possible invalidation reasons enumerated

Reply:
After looking at this again, some places prepend data to the string to give more context. Because of this it doesn't work nicely.